### PR TITLE
records: refactor receivers

### PR DESCRIPTION
**Pull request by jacquerie**
Sentry: https://sentry.cern.ch/inspire-sentry/inspire-labs-qa/group/601665/

### CODE REVIEW

File: inspirehep/modules/records/receivers.py [hunk 1]
```diff
@@ -75,30 +84,74 @@ def normalize_field_categories(sender, *args, **kwargs):
...
+        table then add the corresponding value to the document types.
+        All values found are added.
+
+          + -------- + -------- +
+          | key      | value    |
+          + -------- + -------- +
+          | lectures | lectures |
+          | review   | review   |
+          + -------- + -------- +
+
```
Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: This is the currently implemented algorithm for assigning a doctype. It doesn't make a lot of sense. What's the real algorithm to implement here?

File: inspirehep/modules/records/receivers.py [hunk 2]
```diff
@@ -75,122 +87,163 @@ def normalize_field_categories(sender, *args, **kwargs):
...
+            result.append(INCLUSIVE_DOC_TYPE_MAP[collection])
+
+    json['facet_inspire_doc_type'] = result
+
+
+def match_valid_experiments(sender, json, *args, **kwargs):
+    """TODO."""
+    def _normalize(experiment):
+        try:
+            result = EXPERIMENTS_MAP[experiment.lower().replace(' ', '')]
```
Comment (mihaibivol) [inline inspirehep/modules/records/receivers.py]: This seems like in most of the cases does just `experiment.lower().replace(' ', '').upper()` (at least for most of the experiments in the map)

Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: Actually for all of them, considering how they were generated...

Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: Uh, no, that's completely wrong.

**Comment by jacquerie** [comment]
Ready for review.

File: inspirehep/modules/records/receivers.py [hunk 3]
```diff
@@ -75,136 +90,195 @@ def normalize_field_categories(sender, *args, **kwargs):
...
         if date_key in json:
             valid_date = create_valid_date(json[date_key])
             if valid_date != json[date_key]:
                 current_app.logger.warning(
-                    'MALFORMED: {0} value in {1}: {3}'.format(
-                        date_key, recid, json[date_key]
-                    )
-                )
+                    'MALFORMED: %s value in %s: %s', date_key,
+                    json['control_number'], json[date_key])
```
Comment (jacquerie) [inline inspirehep/modules/records/receivers.py]: @mihaibivol, this is ok, right?

**Comment by jacquerie** [comment]
Still ready for review, any takers?

File: inspirehep/modules/records/receivers.py (hunk 3 above)
Comment (mihaibivol) [inline inspirehep/modules/records/receivers.py]: Yup.

**Pull request merged**